pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/protobuf/CodecParseMethodGenerator.java

```diff
@@ -74,33 +98,33 @@ static String generateParseMethod(final String modelClassName, final List<Field>
                             // So we catch this exception here and **only** here, because an EOFException
                             // anywhere else suggests that we're processing malformed data and so
                             // we must re-throw the exception then.
-                            final int tag;
+                            final int $prefixtag;
```

Comment:
Seems a bit confusing if I am reading this right. We so far have been using `$xxxx` as placeholders for replacement in strings. Looks like you are now introducing using `$` prefixed variables in generated code as well. This makes the code generation harder to read in my opinion as you do not know if it is a runtime variable or a string substitution.

Reply:
You're misreading this part a bit. Here I'm introducing a placeholder named `$prefix` that is being `replace()`'d at line 154 below. So the `$prefixtag` is in fact a concatenation of the `$prefix` value replaced at line 154 and a literal string `tag`. So this code doesn't introduce any actual variable names starting with `$` in the actual generated code - they all get replaced.

Note that if the `$prefix` value at line 154 happens to be an empty string, then this particular variable will be named just `tag` as it's always been previously. The reason I need to introduce the `$prefix` is to support nesting the parsing code. This is necessary because, as stated elsewhere, for performance reason we don't model map entries as actual messages, and hence we cannot reuse the regular message parsing code path which would delegate to the message's PROTOBUF.parse() method. Instead we simply nest the map parsing loop inside the main parsing loop. But this creates a name conflict for Java variables such as this `tag` and javac simply fails (even though this nested loop is a separate `{}` block - it still won't let me overload a local variable name, and I think for a good reason.) So I just added this `$prefix` to resolve the conflict.

However, I agree that this looks a bit confusing, and the confusion mainly arises from the fact that "prefixtag" reads as a single word from human perspective while in fact it's two separate tokens from this code perspective.

If you wish, I could change the case of the prefix and change this to e.g. `$PREFIXtag`, but I'm unsure if this is too very helpful.

For now, I'm leaving this as is. But if you have any concrete suggestion on how to improve this part, I'm happy to implement it. Please let me know.